src/helpers/format-chat-history.ts

```diff
@@ -49,6 +50,26 @@ export async function formatChatHistory(
   return Array.from(new Set(chatHistory)).filter((x): x is string => !!x);
 }
 
+export async function formatSpecAndPull(context: Context<"pull_request.opened" | "pull_request.ready_for_review">, issue: Issue): Promise<string[]> {
+  const tokenLimits: TokenLimits = {
+    modelMaxTokenLimit: context.adapters.openai.completions.getModelMaxTokenLimit(context.config.openAiModel),
+    maxCompletionTokens: context.config.maxTokens || context.adapters.openai.completions.getModelMaxOutputLimit(context.config.openAiModel),
+    runningTokenCount: 0,
+    tokensRemaining: 0,
+  };
+
+  // what we start out with
+  tokenLimits.tokensRemaining = tokenLimits.modelMaxTokenLimit - tokenLimits.maxCompletionTokens;
+
+  const chatHistory = await createPullSpecContextBlockSection({
+    context,
+    tokenLimits,
+    issue,
+  });
+
+  return Array.from(new Set([chatHistory])).filter((x): x is string => !!x);
```

Comment:
`chatHistory` is a string so why all this stuff?
Also `chatHistory` seems like a weird name for spec and diff